RHDEVDOCS-3750+3704 Tekton Hub with OpenShift Pipelines (#43307)
openshift/openshift-docs: modules/op-setting-a-cron-job-for-refreshing-catalog-in-tekton-hub.adoc

// This module is included in the following assembly:
//
// *cicd/pipelines/using-tekton-hub-with-openshift-pipelines.adoc

:_content-type: PROCEDURE
[id="setting-a-cron-job-for-refreshing-catalog-in-tekton-hub_{context}"]
= Optional: Setting a cron job for refreshing catalog in {tekton-hub}

[role="_abstract"]

> Cluster administrators can optionally set up a cron job to refresh the database after a fixed interval, so that changes in the catalog reflects in the UI.

Reviewer: ```suggestion
Cluster administrators can optionally set up a cron job to refresh the database after a fixed interval so that changes in the catalog reflects in the UI.
```

[NOTE]
====

> If new resources are added or existing resources are updated in the catalog, a catalog refresh action displays it in the UI. However, if a resource is deleted in the catalog, a catalog refresh action does not delete the resource in the database, and continues to display it in the UI.

Reviewer: ```suggestion
If new resources are added, or existing resources are updated in the catalog, refreshing the catalog displays these changes in the <specify which UI>. However, if a resource is deleted from the catalog, refreshing the catalog does not remove the resource from the database. The <name of UI> continues displaying the deleted resource.
```

Author: @PuneetPunamiya - Rolfe raises a valid point here. Can you please let me know what happens if a user attempts using a resource that has been deleted from the catalog, but still persists in the backend database?

====

[discrete]
.Prerequisites
* Ensure that you are in the `<project_root>/config/` directory, where `<project_root>` is the top level directory of the cloned {tekton-hub} repository.
* Ensure that you have a JWT token with a scope of refreshing the catalog.

[discrete]
.Procedure
. Create an agent-based JWT token for longer use.
+
[source,terminal]
----
$ curl -X PUT --header "Content-Type: application/json" \
    -H "Authorization: <access-token>" \ <1>
    --data '{"name":"catalog-refresh-agent","scopes": ["catalog:refresh"]}' \
    <api-route>/system/user/agent
----

> <1> Replace the `<access-token>` with the JWT token.

Reviewer: Do we need to spell out what JWT means?

+
The agent token with the necessary scopes are returned in the `{"token":"<agent_jwt_token>"}` format. Note the returned token and preserve it for the catalog refresh cron job.

. Edit the `05-catalog-refresh-cj/50-catalog-refresh-secret.yaml` file to set the `HUB_TOKEN` parameter to the `<agent_jwt_token>` returned in the previous step.
+
[source,yaml]
----
apiVersion: v1
kind: Secret
metadata:
  name: catalog-refresh
type: Opaque
stringData:
  HUB_TOKEN: <hub_token> <1>
----
<1> Replace with the `<agent_jwt_token>` returned in the previous step.

. Apply the modified YAML files.
+
[source,terminal]
----
$ oc apply -f 05-catalog-refresh-cj/ -n openshift-pipelines.
----

> . (Optional) By default, the cron job is configured to run every 30 minutes. To change the interval, modify the value of the `schedule` parameter in the `05-catalog-refresh-cj/51-catalog-refresh-cronjob.yaml` file.

Reviewer: ```suggestion
. Optional: By default, the cron job is configured to run every 30 minutes. To change the interval, modify the value of the `schedule` parameter in the `05-catalog-refresh-cj/51-catalog-refresh-cronjob.yaml` file.
```
Multiple instances. Use the standard formatting for optional steps.
